proposed/http-client/http-client.md

```diff
@@ -46,6 +46,11 @@ interface HttpClient
      *
      * If a request is sent without any prior configuration, a HTTP client MUST NOT follow redirects.
      *
+     * The client MAY do motifications to the Request before sending it. Because PSR-7 objects are
+     * immutable, one cannot assume that the object passed to HttpClient::sendRequest will be the same
+     * object that is actually sent. For example the Request object that is returned by an exception MAY
+     * be a different object than the one passed to sendRequest, so === comparison is not possible.
```

Comment:
@dbu whats to move away from the word "reference" (https://github.com/php-http/fig-standards/pull/16#discussion_r80212953). I have no issues using that word as long as it makes sense and is easy to understand.